lib/model/store.dart

```diff
@@ -190,10 +208,27 @@ abstract class GlobalStore extends ChangeNotifier {
   /// Update an account in the underlying data store.
   Future<void> doUpdateAccount(int accountId, AccountsCompanion data);
 
+  /// Remove an account from the store.
+  Future<void> removeAccount(int accountId) async {
+    await doRemoveAccount(accountId);
+    assert(_accounts.containsKey(accountId));
+    _accounts.remove(accountId);
+    _perAccountStores.remove(accountId)
+      // TODO close connection inside `.dispose` instead (once tests can adapt)
+      ?..dispose()..connection.close();
+    _perAccountStoresLoading.remove(accountId);
```

Comment:
Even though this is logic I largely suggested a week or two ago, I found it kind of confusing to read. :slightly_smiling_face: I think the logic is good but would benefit from some comments.

I can take a pass at writing those comments in the next round.

Reply:
Cool; thanks.